src/Microsoft.Extensions.Options/OptionsServiceCollectionExtensions.cs

```diff
@@ -27,7 +27,10 @@ public static IServiceCollection AddOptions(this IServiceCollection services)
             services.TryAdd(ServiceDescriptor.Singleton(typeof(IOptions<>), typeof(OptionsManager<>)));
             services.TryAdd(ServiceDescriptor.Scoped(typeof(IOptionsSnapshot<>), typeof(OptionsManager<>)));
             services.TryAdd(ServiceDescriptor.Singleton(typeof(IOptionsMonitor<>), typeof(OptionsMonitor<>)));
-            services.TryAdd(ServiceDescriptor.Singleton(typeof(IOptionsFactory<>), typeof(OptionsFactory<>)));
+            services.TryAdd(ServiceDescriptor.Transient(typeof(IOptionsService<>), typeof(OptionsService<>)));
+            services.TryAdd(ServiceDescriptor.Transient(typeof(IOptionsFactory<>), typeof(OptionsFactory<>)));
+            services.TryAdd(ServiceDescriptor.Transient(typeof(IOptionsValidator<>), typeof(OptionsValidator<>)));
```

Comment:
I think that IConfigure/Validates should be stateless. And if they stateless then it's fine if they are singletons.

Why it might be required to have IConfigure/Validates with different lifetimes? Configure by values from HttpContext? (sounds ugly)